docs/content/Deployment/Overview.md

```diff
@@ -0,0 +1,288 @@
+
+      - CUBEJS_API_SECRET=secret
+    volumes:
+      - .:/cube/conf
+    depends_on:
+      - cubestore_router
+      - cube_refresh_worker
+      - redis
+
+  cube_refresh_worker:
+    image: cubejs/cube
+    environment:
+      - CUBEJS_DB_TYPE=bigquery
+      - CUBEJS_DB_BQ_PROJECT_ID=cubejs-k8s-cluster
+      - CUBEJS_DB_BQ_CREDENTIALS=<BQ-KEY>
+      - CUBEJS_DB_BQ_EXPORT_BUCKET=cubestore
+
+      - CUBEJS_CUBESTORE_HOST=cubestore_router
+
+      - CUBEJS_REDIS_URL=redis://redis:6379
+
+      - CUBEJS_API_SECRET=secret
+
+      - CUBEJS_SCHEDULED_REFRESH_TIMER=true
+    volumes:
+      - .:/cube/conf
+
+  cubestore_router:
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_WORKERS=cubestore_worker_1:10001,cubestore_worker_2:10002
+      - CUBESTORE_REMOTE_DIR=/cube/data
+      - CUBESTORE_META_PORT=9999
+      - CUBESTORE_SERVER_NAME=cubestore_router:9999
+    volumes:
+      - .cubestore:/cube/data
+    depends_on:
+      - cubestore_worker_1
+      - cubestore_worker_2
+      - cubestore_worker_3
+      - cubestore_worker_4
+
+  cubestore_worker_1:
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_SERVER_NAME=cubestore_worker_1:10001
+      - CUBESTORE_WORKER_PORT=10001
+      - CUBESTORE_REMOTE_DIR=/cube/data
+      - CUBESTORE_META_ADDR=cubestore_router:9999
+    volumes:
+      - .cubestore:/cube/data
+
+  cubestore_worker_2:
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_SERVER_NAME=cubestore_worker_2:10002
+      - CUBESTORE_WORKER_PORT=10002
+      - CUBESTORE_REMOTE_DIR=/cube/data
+      - CUBESTORE_META_ADDR=cubestore_router:9999
+    volumes:
+      - .cubestore:/cube/data
+
+  redis:
+    image: bitnami/redis:latest
+    environment:
+      - ALLOW_EMPTY_PASSWORD=yes
+    logging:
+      driver: none 
+
+```
+
+
+## API Instance
+
+API instance is processing incoming API requests and quering either Cube Store for pre-aggregated data or connected database(s) for raw data. It is possible to horizontally scale API instances and use load balancer to balance incoming requests between multiple API instances.
+
+[Cube.js docker image](https://hub.docker.com/r/cubejs/cube) is used for API Instance.
+
+API instance needs to be configured via environment variables, cube.js file and
+has access to the data schema files.
+
+## Refresh Worker
+
+Refresh Worker updates the pre-aggregations and in-memory cache in the
+background.
+
+[Cube.js docker image](https://hub.docker.com/r/cubejs/cube) is used for Refresh Worker too.
+To make service act as a Refresh Worker `CUBEJS_SCHEDULED_REFRESH_TIMER=true` should be set.
+
+## Cube Store
+
+Cube Store is the purpose-built pre-aggregations storage for Cube.js.
+
+Cube Store uses a distributed query engine architecture. In every Cube Store cluster:
+
+* a one or many router nodes handle incoming connections, manages database metadata, builds query plans, and orchestrates their execution
```

Comment:
Love the highlander reference! ❤️ 

I remember @paveltiunov mentioned potential replication of the router nodes. We need to confirm whether this replication would be inside the single router or It would just scale by adding more routers and load balance requests.